components/channel/include/channel.h

```diff
@@ -0,0 +1,211 @@
+
+/**
+ * channel_init_input - helper function to initialize channel input object
+ * @ch: pointer to channel object to initialize
+ * @identifier: identifier string of the channel
+ * @queue: queue handle to send input to 
+ */
+static
+void
+inline __attribute__((always_inline))
+channel_init_input
+(Channel_in *ch, const char *identifier, QueueHandle_t queue)
+{
+    return channel_init(ch, identifier, queue, queueSEND_TO_BACK, &xQueueGenericSend);
+}
+
+/**
+ * channel_init_output - helper function to initialize channel output object
+ * @ch: pointer to channel object to initialize
+ * @identifier: identifier string of the channel
+ */
+static
+void
+inline __attribute__((always_inline))
+channel_init_output
+(Channel_out *ch, const char *identifier)
+{
+    return channel_init(ch, identifier, NULL, 0, NULL);
+}
+
+/**
+ * channel_setContext - helper function to change the context the callback acts on
+ * @ch: pointer to channel object to modify
+ * @ctx: pointer to new context
+ */
+static
+void
+inline __attribute__((always_inline))
+channel_setContext 
+(Channel *ch, void *ctx)
+{
+    ch->ctx = ctx;
+}
+
+/**
+ * channel_setCallback - helper function to change the callback
+ * @ch: pointer to channel object to modify
+ * @callback: function to register as callback
+ */
+static
+void 
+inline __attribute__((always_inline))
+channel_setCallback
+(Channel *ch, const Channel_callback callback)
+{
+    ch->callback = callback;
+}
+
+/**
+ * reset_channel - removes a channel input or output from the message structure
+ * @ch: channel element to reset
+ */
+void
+channel_reset
+(Channel *ch);
+
+/**
+ * channel_send - sends a message to a single channel object
+ * @ch: the channel structure that holds informations for this channel
+ * @data: the pointer to the message
+ * @timeout: how many time-slices the callback may block before aborting
+ * @return: returns pdPass (=1) if send was successful, if non-one value is returned, callback failed
+ * 
+ * Calls the callback function on the channel.
+ */
+BaseType_t
+channel_send
+(const Channel_in *ch, const void *data, const TickType_t timeout);
+
+/**
+ * channel_broadcast_init - helper function to initialize broadcast handler
+ * @handle: pointer to broadcast object
+ * @ch: pointer to channel object initiating the broadcast
+ * @data: pointer to data to be transferred
+ * @timeout: number of time-slices each receiver may block 
+ */
+static
+void
+inline __attribute__((always_inline))
+channel_broadcast_init
+(Channel_broadcast *handle, const Channel_out *ch, const void *data, const TickType_t timeout)
+{
+    handle->ch = (void*) ch;
+    handle->pos = (void*) ch;
+    handle->data = (void*) data;
+    handle->timeout = timeout;
+}
+
+/**
+ * channel_broadcast_finished - helper function to determine if broadcast finished
+ * @ch: pointer to channel object
+ * @pos: location of pointer where broadcast stopped 
+ */
+static
+bool
+inline __attribute__((always_inline))
```

Comment:
as far as i understand \_\_attribute\_\_((always_inline)) prepending inline is not needed
```suggestion
__attribute__((always_inline))
```